### PR TITLE
Fixes and more fixes

### PR DESCRIPTION
**Pull request by Sxyntheon**

### CODE REVIEW

File: backend/src/api_wrapper/untis_client.rs [hunk 1]
```diff
@@ -114,7 +114,7 @@ impl UntisClient {
     }
 
     async fn get_ids(&mut self) -> Result<HashMap<String, u16>, Error> {
-        let klassen: Vec<Klasse> = self.get_klassen().await.map_err(|err| Error::UntisError(err.to_string() + " 118"))?;
+        let klassen: Vec<Klasse> = self.get_klassen().await.map_err(|err| Error::UntisError(err.to_string()))?;
```
Comment (ixhbinphoenix) [inline backend/src/api_wrapper/untis_client.rs:117]: why

File: frontend/src/api/dateHandling.ts [hunk 2]
```diff
@@ -82,12 +82,14 @@ function isBetweenDates(startDateStr: string, endDateStr: string, todayDate: Dat
   const startDate = new Date(
     parseInt(startDateStr.slice(0, 4), 10),
     parseInt(startDateStr.slice(4, 6), 10) - 1,
-    parseInt(startDateStr.slice(6, 8), 10)
+    parseInt(startDateStr.slice(6, 8), 10),
+    2
```
Comment (ixhbinphoenix) [inline frontend/src/api/dateHandling.ts:86]: why/how does this work?

File: backend/src/api_wrapper/untis_client.rs (hunk 1 above)
Comment (Sxyntheon) [reply backend/src/api_wrapper/untis_client.rs:117]: Cause I need the error messages to be the same in all files

Comment (ixhbinphoenix) [reply backend/src/api_wrapper/untis_client.rs:117]: ah yeah makes sense

File: frontend/src/api/dateHandling.ts (hunk 2 above)
Comment (Sxyntheon) [reply frontend/src/api/dateHandling.ts:86]: Goofy Zeitumstellung.
Aus 2024-05-24:00
Wird 2024-05-23:22

**Review by ixhbinphoenix — APPROVED** [review]

**Pull request merged**